### PR TITLE
AzureADB2C: make it possible to define multiple policies

### PR DESCRIPTION
**Pull request by lsmith77**
fixes https://github.com/SocialiteProviders/Providers/issues/818

This all feels quite hacky, especially overriding the `redirectUrl` but I didn't manage to figure out another way.

Note I am using SocialStream, so I had to override quite a few things there as well, to pass the policy as a parameter to the `Provider`

```
Socialite::driver($provider)
  ->with(['policy' => $policy])
  ->redirect();
```

I hope I sufficiently taken care of backwards compatibility.

### CODE REVIEW

File: src/AzureADB2C/Provider.php [hunk 1]
```diff
@@ -148,7 +187,6 @@ protected function mapUserToObject(array $user)
     {
         return (new User())->setRaw($user)->map([
             'id'   => $user['sub'],
-            'name' => $user['name'],
```
Comment (lsmith77) [inline src/AzureADB2C/Provider.php:151]: this should be left to the user landcode to determine what should be mapped

**Comment by lsmith77** [comment]
@fujie what do you think about this change?

**Comment by fujie** [comment]
@lsmith77 Let me understand what you're trying to do. Is it correct that you want to use multiple policies for the same application by overriding policy and redirect_uri when calling the driver, and you're proposing to change default claim mapping on Provider.php to map only sub claim to override the claims mapping for every policies?
If I understand correctly, I think it is reasonable change.

**Comment by lsmith77** [comment]
yeah exactly!

**Comment by lsmith77** [comment]
@lucasmichot given that @fujie agrees with the change, are you willing to merge this PR?

**Comment by lucasmichot** [comment]
> @lucasmichot given that @fujie agrees with the change, are you willing to merge this PR?

We typically prefer having a double review @lsmith77 , as soon as an other member will review 🟢  this PR, it will be merged.

**Review by lucasmichot — APPROVED** [review]
Thanks a lot for the PR @lsmith77

**Comment by lsmith77** [comment]
ok thank you. Will see if I can get another person to review.

f.e. I have now also implemented a way to get an `access_token`. but not sure if this is sufficiently generic and definately goes beyond the scope of this PR. just leaving it here for now.

```
    public function getAccessTokenResponse($code)
    {
        $fields = $this->getTokenFields($code);
        $fields['scope'] = implode(' ', $this->getScopes());

        $response = $this->getHttpClient()->post($this->getTokenUrl(), [
            RequestOptions::FORM_PARAMS => $fields,
        ]);

        return json_decode((string) $response->getBody(), true);
    }
```

**Review by atymic — APPROVED** [review]
LGTM 👍

**Pull request merged**